File: attacus/flutter_config.py
from pathlib import Path
from importlib import resources

from loguru import logger

from . import attacus as core
from .assets import get_assets_path
from .artifacts import get_icu_data_path

class FlutterConfig(core.FlutterConfig):
    def __init__(self) -> None:
        super().__init__()
        #self.assets_path = str(Path.cwd() / 'flutter_assets')
        self.assets_path = str(get_assets_path())
        self.icu_data_path = str(get_icu_data_path())
        #logger.debug(self.icu_data_path)

File: src/attacus/flutter/flutter_config.cpp
#include <filesystem>
namespace fs = std::filesystem;

#include "flutter_config.h"

namespace attacus {
    FlutterConfig::FlutterConfig() {
        fs::path project_path = fs::current_path();
        fs::path assets_path = project_path / "build" / "flutter_assets";
        fs::path icu_data_path = project_path / "build" / "icudtl.dat";
        assets_path_ = assets_path.string();
        icu_data_path_ = icu_data_path.string();
    }
}